crmsh/bootstrap.py

```diff
@@ -104,6 +105,179 @@ def init_qdevice(self):
                 cmds=self.qdevice_heuristics,
                 mode=self.qdevice_heuristics_mode)
 
+    def init_sbd_manager(self):
+        self.sbd_manager = SBDManager(self.sbd_device, self.diskless_sbd)
+
+
+class SBDManager(object):
+    """
+    Class SBDManager is for sbd configuration and sbd service management
+    """
+    sbd_config_dict = {
+            "SBD_PACEMAKER": "yes",
+            "SBD_STARTMODE": "always",
+            "SBD_DELAY_START": "no"
+            }
+
+    def __init__(self, sbd_device=None, diskless_sbd=False):
+        """
+        Parameter sbd_device is provided by '-s' option on init process
+        Parameter diskless_sbd is provided by '-S' option on init process
+        """
+        self.sbd_device = sbd_device
+        self.diskless_sbd = diskless_sbd
+        self._sbd_service_flag = False
+
+    @property
+    def sbd_service_flag(self):
+        return self._sbd_service_flag
+
+    @sbd_service_flag.setter
+    def sbd_service_flag(self, value):
+        """
+        Set flag to enable sbd service on join process
+        """
+        self._sbd_service_flag = value
+
+    def sbd_init(self):
+        """
+        Function sbd_init includes these steps:
+        1. Check local environment
+        2. Get sbd device from options or interactive mode
+        3. Initialize sbd device
+        4. Write config file /etc/sysconfig/sbd
+        """
+        self._check_environment()
+        self._get_sbd_device()
+        if not self.sbd_device and not self.diskless_sbd:
+            return
+        self._initialize_sbd()
+        self._update_configuration()
+        # If process work through here, consider it's ready for enable service
+        self._sbd_service_flag = True
+
+    def configured_sbd_device(self):
```

Comment:
I would change the name of this method to `get_sbd_configuration`

Reply:
Changed